scanpy/tools/rank_genes_groups.py

```diff
@@ -181,19 +181,36 @@ def rank_genes_groups(
             rankings_gene_scores.append(scores[global_indices])
             rankings_gene_names.append(adata_comp.var_names[global_indices])
     elif method == 'logreg':
+    #if reference is not set, then the groups listed will be compared to the rest
+    #if reference is set, then the groups listed will be compared only to the other groups listed
         from sklearn.linear_model import LogisticRegression
-        if reference != 'rest':
-            raise ValueError('\'logreg\' is only implemented for `reference==\'rest\'`.')
+        if reference not in groups:
+            raise Exception("reference must be in groups")
+        if len(groups) == 1:
+            raise Exception("Cannot perform logistic regression on a single cluster.")
+        if len(groups) == 2: #binary logistic regression
+            adata_copy = adata[(adata.obs[groupby] == groups_order[0]) | (adata.obs[groupby] == groups_order[1])]
+            X = adata_copy.raw.X
```

Comment:
What if use_raw is False?

Reply:
very right... i'll go over and correct it...

Reply:
If you haven't yet, I can go fix it... sorry about that